lib/enums/login_status.dart

```diff
@@ -0,0 +1 @@
+enum LoginStatus { Initial, Loading, Success, Failure }
```

Comment:
Have a look at
- freezed package
- dart 3 sealed classes

You can use either on of these to model your state